### PR TITLE
[VUMM-791] Permission v2 changes for Project Service

### PR DESCRIPTION
**Pull request by anandjakhaniya**
## Testing
- [ ] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

## Reverting
- [ ] Contains Migration - _Do Not Revert_

### CODE REVIEW

**Review by evgeniilezhnin — APPROVED** [review]

File: backend/server/src/main/java/ai/verta/modeldb/config/MDBConfig.java [hunk 1]
```diff
@@ -27,6 +27,7 @@ public class MDBConfig extends Config {
   @JsonProperty private TelemetryConfig telemetry;
   @JsonProperty private List<MigrationConfig> migrations;
   @JsonProperty protected FutureJdbi jdbi;
+  @JsonProperty private boolean enabledPermissionV2 = false;
```
Comment (eheinlein-verta) [inline backend/server/src/main/java/ai/verta/modeldb/config/MDBConfig.java]: I think this reads better with the name `permissionV2Enabled`

File: backend/server/src/main/java/ai/verta/modeldb/batchProcess/DatasetToRepositoryMigration.java [hunk 2]
```diff
@@ -71,6 +71,7 @@ private DatasetToRepositoryMigration() {}
   public static void execute(int recordUpdateLimit) {
     DatasetToRepositoryMigration.recordUpdateLimit = recordUpdateLimit;
     config = App.getInstance().mdbConfig;
+    var executor = FutureExecutor.initializeExecutor(config.getGrpcServer().getThreadCount());
```
Comment (eheinlein-verta) [inline backend/server/src/main/java/ai/verta/modeldb/batchProcess/DatasetToRepositoryMigration.java]: Is this change related to the main purpose of the PR, or is it just cleanup? (Fine either way, I just want to check.)

**Review by eheinlein-verta — APPROVED** [review]
One change requested (config name) - but that doesn't have to be a blocker for merging.

File: backend/server/src/main/java/ai/verta/modeldb/config/MDBConfig.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/server/src/main/java/ai/verta/modeldb/config/MDBConfig.java]: Renamed

File: backend/server/src/main/java/ai/verta/modeldb/batchProcess/DatasetToRepositoryMigration.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/server/src/main/java/ai/verta/modeldb/batchProcess/DatasetToRepositoryMigration.java]: Reverted this change. it was not needed.

**Comment by anandjakhaniya** [comment]
!rebuild

**Review by evgeniilezhnin — CHANGES_REQUESTED** [review]
please fix mocking errors

**Comment by evgeniilezhnin** [comment]
test runs have now different errors about void mocks, try to fix them

**Review by evgeniilezhnin — APPROVED** [review]
seems that it is working after fixes in the next PRs

**Pull request merged**